### ISSUE TITLE
Calling object method that doesn't return using extended variable syntax

### ISSUE THREAD

**Issue by tw39124-1** [robotframework/robotframework#3978]
If I call a method of an object in extended variable syntax, and that method doesn't return anything (i.e. it just goes and 'does' something):

```
${TARGET.power_off()}
```
then I get a "Keyword name cannot be empty" error and in the log file it looks like Robot thinks I am trying to assign something to that variable but with no actual keyword:

```
${TARGET.power_off()} =
```

The examples of extended variable syntax in the user guide only show use cases where the method returns a value which is used by another keyword.  Is there a nice way around this without having to resort to using "Call Method" everywhere ?? Perhaps some syntax I don't know about that I can use to allow me to just do `object.method()` in a simple way?

**Comment by pekkaklarck** [comment]
Extended variable syntax is to be used in places where you in general use variables, it isn't designed to be used like keywords to execute actions. Your options include:

1. Use it with some keyword that doesn't itself have side-effects but does accept arguments like `Log    ${TARGET.power_off()`. This looks pretty odd and I don't really recommend it.
2. Use `Call Method`, as you already suggested, like `Call Method    ${TARGET}    power_off`. Better than the above but I don't like this too much either.
3. Create a library that handles this. It could either contain the context info internally so that you can use it like `Power Off` or it could require passing the target to operate as an argument like `Power Off    ${TARGET}`: I generally like the former approach more as it typically makes actual test/task data simpler. Either way, this is the generally recommended approach.

In other words, you should create a library instead. If you consider that an overkill solution for your need, you can use `Call Method`. I see no needs to add more ways to handle this situation.
